packages/adapter-edgedb/src/index.ts

```diff
@@ -0,0 +1,258 @@
+import type { Adapter } from "next-auth/adapters";
+import type { Client } from "edgedb";
+
+import e from "../dbschema/edgeql-js";
+
+export function EdgeDBAdapter(client: Client): Adapter {
+  return {
+    async createUser(user) {
+      const query = e.select(
+        e.insert(e.User, {
+          email: user.email,
+          emailVerified: user.emailVerified,
+          name: user.name,
+          image: user.image,
+        }),
+        () => ({
+          id: true,
+          email: true,
+          emailVerified: true,
+          name: true,
+          image: true,
+        })
+      );
+
+      return await query.run(client);
+    },
+    async getUser(id) {
+      const query = e.select(e.User, () => ({
+        id: true,
+        email: true,
+        emailVerified: true,
+        name: true,
+        image: true,
+        filter_single: { id },
+      }));
+
+      return await query.run(client);
+    },
+    async getUserByEmail(email) {
+      const query = e.select(e.User, () => ({
+        id: true,
+        email: true,
+        emailVerified: true,
+        name: true,
+        image: true,
+        filter_single: { email },
+      }));
+
+      return await query.run(client);
+    },
+    async getUserByAccount({ providerAccountId }) {
+      const query = e
+        .select(e.User, (user) => ({
+          ...e.User["*"],
+          filter_single: e.op(
+            user.accounts.providerAccountId,
+            "=",
+            providerAccountId
+          ),
+        }))
+        .assert_single();
+
+      return await query.run(client);
+    },
+    async updateUser({ email, emailVerified, id, image, name }) {
+      const updatedUser = await e
+        .update(e.User, () => ({
+          filter_single: { id: String(id) },
+          set: {
+            ...(email && { email }),
+            ...(emailVerified && { emailVerified }),
+            ...(image && { image }),
+            ...(name && { name }),
+          },
+        }))
+        .run(client);
+
+      if (!updatedUser) {
+        throw new Error("could not update user");
+      }
+
+      const user = await e
+        .select(e.User, () => ({
+          id: true,
+          email: true,
+          emailVerified: true,
+          image: true,
+          name: true,
+          filter_single: {
+            id: updatedUser.id,
+          },
+        }))
+        .run(client);
```

Comment:
You can wrap the `update` query above in this `select` to update and select in a single query.